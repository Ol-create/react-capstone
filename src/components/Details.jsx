import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleLeft } from '@fortawesome/free-regular-svg-icons';
import { coinDetail } from '../redux/CoinState/detailReducer';

const Details = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const cryto = useSelector((store) => store.detail);
  useEffect(() => {
    dispatch(coinDetail(id));
  }, []);
const { symbol, name:title, rank, priceUsd, volumeUsd24Hr, supply } = cryto;
  return (
    <section>
      <NavLink to="/">
        <FontAwesomeIcon icon={faCircleLeft} size="xl" className="back" />
      </NavLink>

      <div className="detail">
        <h2>{symbol}</h2>
        <p>{title}</p>
      </div>
      <div className="title">
        Details
      </div>
      <div className="detailList">
        <div>
          <p>Rank</p>
        </div>
        <div>
          <p>
            {' #'}
            {rank}
          </p>
        </div>
        <div className="dark">
          <p>Price</p>
        </div>
        <div className="dark">
          <p>
            {' $'}
            {priceUsd}
            {' USD'}
          </p>
        </div>
        <div>
          <p>Volume 24h</p>
        </div>
        <div>
          <p>
            {' $'}
            {volumeUsd24Hr}
          </p>
        </div>
        <div className="dark">
          <p>Total Supply</p>
        </div>
        <div className="dark">
          <p>
            {' '}
            {supply}
            {' '}
            {symbol}
          </p>
        </div>
      </div>
    </section>
  );
};

export default Details;
