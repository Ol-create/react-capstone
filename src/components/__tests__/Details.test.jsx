import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/configureStore';
import Details from '../Details';
import { coinDetail } from '../../redux/CoinState/detailReducer';

describe('Details component', () => {
  test('component match with the snapshot', () => {
   store.dispatch(coinDetail('bitcoin'));
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
