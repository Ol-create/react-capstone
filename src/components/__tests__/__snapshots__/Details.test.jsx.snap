// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component component match with the snapshot 1`] = `
<section>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-left fa-xl back"
      data-icon="circle-left"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M360 224L272 224v-56c0-9.531-5.656-18.16-14.38-22C248.9 142.2 238.7 143.9 231.7 150.4l-96 88.75C130.8 243.7 128 250.1 128 256.8c.3125 7.781 2.875 13.25 7.844 17.75l96 87.25c7.031 6.406 17.19 8.031 25.88 4.188s14.28-12.44 14.28-21.94l-.002-56L360 288C373.3 288 384 277.3 384 264v-16C384 234.8 373.3 224 360 224zM256 0C114.6 0 0 114.6 0 256s114.6 256 256 256s256-114.6 256-256S397.4 0 256 0zM256 464c-114.7 0-208-93.31-208-208S141.3 48 256 48s208 93.31 208 208S370.7 464 256 464z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <div
    className="detail"
  >
    <h2>
      BTC
    </h2>
    <p>
      Bitcoin
    </p>
  </div>
  <div
    className="title"
  >
    Details
  </div>
  <div
    className="detailList"
  >
    <div>
      <p>
        Rank
      </p>
    </div>
    <div>
      <p>
         #
        1
      </p>
    </div>
    <div
      className="dark"
    >
      <p>
        Price
      </p>
    </div>
    <div
      className="dark"
    >
      <p>
         $
        20280.3679042756291630
         USD
      </p>
    </div>
    <div>
      <p>
        Volume 24h
      </p>
    </div>
    <div>
      <p>
         $
        14055542245.4719429581422542
      </p>
    </div>
    <div
      className="dark"
    >
      <p>
        Total Supply
      </p>
    </div>
    <div
      className="dark"
    >
      <p>
         
        19169768.0000000000000000
         
        BTC
      </p>
    </div>
  </div>
</section>
`;
